.gitlab-ci.yml

```diff
@@ -854,21 +854,23 @@ workflow:
 
 .on_system_probe_changes_or_manual:
   - changes:
-      - pkg/collector/corechecks/ebpf/**/*
-      - pkg/ebpf/**/*
-      - pkg/network/**/*
-      - pkg/util/kernel/**/*
-      - test/kitchen/site-cookbooks/dd-system-probe-check/**/*
-      - test/kitchen/test/integration/system-probe-test/**/*
-      - test/kitchen/test/integration/win-sysprobe-test/**/*
-      - .gitlab/functional_test/system_probe.yml
-      - .gitlab/kernel_version_testing/system_probe.yml
-      - test/new-e2e/system-probe/**/*
-      - test/new-e2e/scenarios/system-probe/**/*
-      - test/new-e2e/runner/**/*
-      - test/new-e2e-utils/**/*
-      - test/new-e2e/go.mod
-      - tasks/system_probe.py
+      paths:
+        - pkg/collector/corechecks/ebpf/**/*
+        - pkg/ebpf/**/*
+        - pkg/network/**/*
+        - pkg/util/kernel/**/*
+        - test/kitchen/site-cookbooks/dd-system-probe-check/**/*
+        - test/kitchen/test/integration/system-probe-test/**/*
+        - test/kitchen/test/integration/win-sysprobe-test/**/*
+        - .gitlab/functional_test/system_probe.yml
+        - .gitlab/kernel_version_testing/system_probe.yml
+        - test/new-e2e/system-probe/**/*
+        - test/new-e2e/scenarios/system-probe/**/*
+        - test/new-e2e/runner/**/*
+        - test/new-e2e-utils/**/*
+        - test/new-e2e/go.mod
+        - tasks/system_probe.py
+      compare_to: main # TODO: use a variable, when this is supported https://gitlab.com/gitlab-org/gitlab/-/issues/369916
```

Comment:
Good to know that we can use this!

Reply:
I still have concern about how will it works if we want to backport changes to a release branch ?

Reply:
I asked Kacper if there was a script ran when branching out the release branch. For now I plan to do the PR myself for one or two cycles and then build some automation if needed. I also have a gitlab PR in review to support env variables https://gitlab.com/gitlab-org/gitlab/-/merge_requests/126860